guides/v2.3/comp-mgr/cli/upgrade-with-plugin.md

```diff
@@ -0,0 +1,195 @@
+
+```bash
+composer show magento/product-community-edition 2.3.* --all | grep -m 1 versions
+```
+
+_Magento Commerce_:
+
+```bash
+composer show magento/product-enterprise-edition 2.3.* --all | grep -m 1 versions
+```
+
+'
+%}
+
+## Modify the Magento updater
+
+_Optional_—If the Magento updater is installed (it is located in `update/`) modify it:
+
+1. Backup `update/` directory.
+2. Create a Composer project.
+
+   _{{ ce }} version 2.3.0:_
+
+    ```bash
+    composer create-project --repository=https://repo.magento.com magento/project-community-edition=2.3.0 temp_dir --no-install
+    ```
+
+    _{{ ee }} version 2.3.0:_
+
+    ```bash
+    composer create-project --repository=https://repo.magento.com magento/project-enterprise-edition=2.3.0 temp_dir --no-install
+    ```
+
+    {:.bs-callout .bs-callout-info}
+      If you need to use a repository that contains non-public packages, such as internal sandboxes, change the URL in `--repository` accordingly.
+3. Remove the old `update/` directory and move the `temp_dir/update/` to the `update/` directory:
+
+   ```bash
+   rm -rf update
+   ```
+
+   ```bash
+   mv temp_dir/update .
+   ```
+
+   ```bash
+   rm -rf temp_dir
+   ```
+
+## Update metadata
+
+{:.bs-callout .bs-callout-info}
+Updating the metadata in `composer.json` file is entirely superficial, not functional.
+
+Update the `"name"`, `"version"`, and `"description"` fields in the `composer.json` file as desired.
+
+## Apply updates
+
+```bash
+composer update
+```
+
+## Clean the Magento cache
+
+After applying an update, you must clean the cache.
+
+```bash
+bin/magento cache:clean
+```
+
+## Manually clear caches and generated content
+
+Clear the `var/` and `generated/` subdirectories:
+
+```bash
+rm -rf var/cache/*
+```
+
+```bash
+rm -rf var/page_cache/*
+```
+
+```bash
+rm -rf generated/code/*
+```
+
+{:.bs-callout .bs-callout-info}
+If you use a cache storage other than the filesystem, such as Redis or Memcached, you must manually clear the cache there too.
+
+## Update the database schema and data
+
+```bash
+bin/magento setup:upgrade
+```
+
+## Disable maintenance mode
+
+```bash
+bin/magento maintenance:disable
+```
+
+## Restart Varnish
+
+_Optional_—If you use Varnish for page caching, restart it:
+
+```bash
+service varnish restart
+```
+
+## Check your work
+
+Open your storefront URL in a web browser to check whether the upgrade was successful. If your upgrade was unsuccessful, your storefront will not load properly.
+
+If the application fails with a  `We're sorry, an error has occurred while generating this email.` error:
+
+1. Reset [file system ownership and permissions] as a user with `root` privileges.
+2. Clear these directories:
+   - `var/cache`
+   - `var/page_cache`
+   - `generated/code`
+3. Check your storefront in your web browser again.
+
+<!-- Link definitions -->
+
+[file system ownership and permissions]: {{ page.baseurl }}/install-gde/prereq/file-system-perms.html
+[Prerequisites]: #prerequisites
```

Comment:
This anchor doesn't exist and is causing link failures in local previews.

Reply:
fixed